jcl/src/java.base/share/classes/openj9/internal/foreign/abi/InternalDowncallHandler.java

```diff
@@ -495,16 +510,31 @@ public MethodHandle getBoundMethodHandle() {
 			MethodType nativeMethodType = methodType(Object.class, Addressable.class, SegmentAllocator.class, long[].class);
 			/*[ENDIF] JAVA_SPEC_VERSION >= 21 */
 
-			MethodHandle boundHandle = lookup.bind(this, "runNativeMethod", nativeMethodType);
+			boundMH = lookup.bind(this, "runNativeMethod", nativeMethodType);
 
 			/* Replace the original handle with the specified types of the C function. */
-			boundHandle = permuteMH(boundHandle, funcMethodType);
-			return boundHandle;
+			boundMH = permuteMH(boundMH, funcMethodType);
+
+			/*[IF JAVA_SPEC_VERSION >= 22]*/
+			return (isFfiProtoOn) ? getNativeMHWithInvokeCache(boundMH) : boundMH;
```

Comment:
Please remove the redundant parentheses.

Reply:
Removed as suggested.